modules/edge-identity/src/identity.rs

```diff
@@ -149,70 +158,29 @@ decl_module! {
 			Ok(())
 		}
 
-		/// Verify an existing identity based on its attested proof. Sender
-		/// be specified on the pre-selected list of verifiers.
-		pub fn verify(origin, identity_hash: T::Hash, vote: bool) -> Result {
+		/// Propose verification to be voted upon by the council
+		pub fn verify_or_deny(origin, identity_hash: T::Hash, approve: bool, verifier_index: usize) -> Result {
 			let _sender = ensure_signed(origin)?;
-
-			ensure!(Self::verifiers().contains(&_sender), "Sender not a verifier");
+			ensure!(verifier_index < Self::verifiers().len(), "Verifier index out of bounds");
+			ensure!(Self::verifiers()[verifier_index] == _sender.clone(), "Sender is not a verifier");
 			let record = <IdentityOf<T>>::get(&identity_hash).ok_or("Identity does not exist")?;
 			match record.stage {
 				IdentityStage::Registered => return Err("No attestation to verify"),
 				IdentityStage::Verified => return Err("Already verified"),
 				IdentityStage::Attested => ()
 			}
 
-			// Check the number of verifications the record has and ensure all are valid
-			let mut valid_verifications: Vec<(T::AccountId, bool)> = [(_sender.clone(), vote)].to_vec();
-			let acct = record.account.clone();
-			if let Some(verifications) = record.verifications {
-				valid_verifications = verifications.clone()
-					.into_iter()
-					.filter(|v| Self::verifiers().contains(&v.0))
-					.collect();
-				valid_verifications.push((_sender.clone(), vote));
-			}
-
-			let yes_votes: Vec<T::AccountId> = valid_verifications.clone()
-				.into_iter()
-				.filter(|v| v.1 == true)
-				.map(|v| v.0)
-				.collect();
-
-			let no_votes: Vec<T::AccountId> = valid_verifications.clone()
-				.into_iter()
-				.filter(|v| v.1 == false)
-				.map(|v| v.0)
-				.collect();
-
-			// Check if we have gather a supermajority of no votes
-			// TODO: What does it mean to fail a verification? If malicious
-			//		 behavior, we should punish the sending account somehow.
-			if no_votes.len() * 3 >= 2 * Self::verifiers().len() {
-				Self::remove_pending_identity(&identity_hash, true);
-				<FrozenAccounts<T>>::mutate(|froze| froze.push(acct));
-				Self::deposit_event(RawEvent::Failed(identity_hash, record.account.into()));
-			} else {
-				// Check if we have gathered a supermajority of yes votes
-				let mut stage = IdentityStage::Attested;
-				let mut expiration_time = record.expiration_time;
-				let registrar = record.account.clone();
-				if yes_votes.len() * 3 >= 2 * Self::verifiers().len() {
-					stage = IdentityStage::Verified;
-					expiration_time = None;
-				}
-				// Insert updated identity record back into the collection
+			if approve {
 				<IdentityOf<T>>::insert(identity_hash, IdentityRecord {
-					stage: stage,
-					expiration_time: expiration_time,
-					verifications: Some(valid_verifications),
+					stage: IdentityStage::Verified,
```

Comment:
So, a single verification is sufficient in this scheme? Makes sense if we're using a council, but not if we're using a verifiers list.